airbyte-db/src/main/java/io/airbyte/db/instance/JobsDatabaseInstance.java

```diff
@@ -0,0 +1,127 @@
+import io.airbyte.db.Databases;
+import io.airbyte.db.ExceptionWrappingDatabase;
+import io.airbyte.db.ServerUuid;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JobsDatabaseInstance implements DatabaseInstance {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JobsDatabaseInstance.class);
+  private static final Function<Database, Boolean> IS_JOBS_DATABASE_CONNECTED = database -> {
+    try {
+      LOGGER.info("Testing jobs database connection...");
+      return database.query(ctx -> ctx.fetchExists(select().from("information_schema.tables")));
+    } catch (Exception e) {
+      return false;
+    }
+  };
+  public static final Function<Database, Boolean> IS_JOBS_DATABASE_READY = database -> {
+    try {
+      LOGGER.info("Testing if jobs database is ready...");
+      Optional<String> uuid = ServerUuid.get(database);
+      return uuid.isPresent();
+    } catch (Exception e) {
+      return false;
+    }
+  };
+
+  private final String username;
+  private final String password;
+  private final String connectionString;
+  private final String schema;
+
+  /**
+   * @param connectionString in the format of
+   *        jdbc:postgresql://${DATABASE_HOST}:${DATABASE_PORT/${DATABASE_DB}
+   */
+  @VisibleForTesting
+  public JobsDatabaseInstance(String username, String password, String connectionString, String schema) {
+    this.username = username;
+    this.password = password;
+    this.connectionString = connectionString;
+    this.schema = schema;
+  }
+
+  public JobsDatabaseInstance(String username, String password, String connectionString) throws IOException {
+    this.username = username;
+    this.password = password;
+    this.connectionString = connectionString;
+    this.schema = MoreResources.readResource("job_tables/schema.sql");
+  }
+
+  @Override
+  public Database getInitialized() {
+    // When we don't need to setup the database, it means the database is initialized
+    // somewhere else, and it is considered ready only when data has been loaded into it.
+    return Databases.createPostgresDatabaseWithRetry(
+        username,
+        password,
+        connectionString,
+        IS_JOBS_DATABASE_READY);
+  }
+
+  /**
+   *
+   */
+  @Override
+  public Database getAndInitialize() throws IOException {
+    // When we need to setup the database, it means the database will be initialized after
+    // we connect to the database. So the database itself is considered ready as long as
+    // the connection is alive.
+    Database database = Databases.createPostgresDatabaseWithRetry(
+        username,
+        password,
+        connectionString,
+        IS_JOBS_DATABASE_CONNECTED);
+
+    new ExceptionWrappingDatabase(database).transaction(ctx -> {
+      boolean hasTables = DatabaseInstance.hasTable(ctx, "airbyte_metadata") &&
+          DatabaseInstance.hasTable(ctx, "jobs") &&
```

Comment:
Feels like we can think of a better place to collect all these names. What about the `DatabaseSchemas` file? It'll force us to update it too.

Reply:
I don't think we can keep all the constants in `DatabaseSchema`. Currently `DatabaseSchema` exists in a downstream module. I can pull this file up to `airbyte-db`. However, this will not work for cloud, whose database schema files will surely live in downstream modules that the `airbyte-db` cannot depend on.

Reply:
Actually what I said is not true. We can always put the an implementation of `DatabaseInstance` and its corresponding `DatabaseSchema` in the same module. I will think about how to refactor `DatabaseSchema`.

Reply:
This is not blocking. I was more thinking out loud :)